frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -302,7 +305,7 @@ pub mod pallet {
                 Error::<T>::AlreadyRegisteredContract
             );
             ensure!(
-                Self::is_contract_valid(&contract_id),
+                SmartContract::is_contract(&contract_id),
```

Comment:
@akru the error is similar after the proposed change
```
error[E0599]: no method named `is_contract` found for enum `traits::SmartContract` in the current scope
   --> frame/dapps-staking/src/pallet/mod.rs:307:33
```

Reply:
I think the issue is that you're still using SmartContract enum in dapps-staking pallet.

It should be moved outside of the pallet and you should only depend on IsContract trait.

E.g. method footprint should be:
        pub fn register(
            origin: OriginFor<T>,
            contract_id: IsContract,
) -> DispatchResultWithPostInfo {


Maybe an additional trait bound is needed since the idea is to use this value as 'id'.